// react-form-wizard-component/dist/react-form-wizard-component.es.js
/*
  * react-form-wizard-component
  * Creator:Parsa Jiravand
  * A react form wizard component with validation and progress bar with no external depenendcies which simplifies tab wizard management.
  * v0.2.7
  * MIT License
  */
 
import J, { useImperativeHandle as cr, useState as Ue, useEffect as je } from "react";
var Oe = { exports: {} }, ue = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var qe;
function or() {
  if (qe)
    return ue;
  qe = 1;
  var F = J, E = Symbol.for("react.element"), O = Symbol.for("react.fragment"), R = Object.prototype.hasOwnProperty, _ = F.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, S = { key: !0, ref: !0, __self: !0, __source: !0 };
  function N(A, h, m) {
    var v, y = {}, T = null, j = null;
    m !== void 0 && (T = "" + m), h.key !== void 0 && (T = "" + h.key), h.ref !== void 0 && (j = h.ref);
    for (v in h)
      R.call(h, v) && !S.hasOwnProperty(v) && (y[v] = h[v]);
    if (A && A.defaultProps)
      for (v in h = A.defaultProps, h)
        y[v] === void 0 && (y[v] = h[v]);
    return { $$typeof: E, type: A, key: T, ref: j, props: y, _owner: _.current };
  }
  return ue.Fragment = O, ue.jsx = N, ue.jsxs = N, ue;
}
var le = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Be;
function vr() {
  return Be || (Be = 1, process.env.NODE_ENV !== "production" && function() {
    var F = J, E = Symbol.for("react.element"), O = Symbol.for("react.portal"), R = Symbol.for("react.fragment"), _ = Symbol.for("react.strict_mode"), S = Symbol.for("react.profiler"), N = Symbol.for("react.provider"), A = Symbol.for("react.context"), h = Symbol.for("react.forward_ref"), m = Symbol.for("react.suspense"), v = Symbol.for("react.suspense_list"), y = Symbol.for("react.memo"), T = Symbol.for("react.lazy"), j = Symbol.for("react.offscreen"), G = Symbol.iterator, H = "@@iterator";
    function X(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = G && e[G] || e[H];
      return typeof r == "function" ? r : null;
    }
    var W = F.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function n(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), i = 1; i < r; i++)
          t[i - 1] = arguments[i];
        ne("error", e, t);
      }
    }
    function ne(e, r, t) {
      {
        var i = W.ReactDebugCurrentFrame, u = i.getStackAddendum();
        u !== "" && (r += "%s", t = t.concat([u]));
        var l = t.map(function(f) {
          return String(f);
        });
        l.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, l);
      }
    }
    var P = !1, Z = !1, Q = !1, C = !1, z = !1, $;
    $ = Symbol.for("react.module.reference");
    function d(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === R || e === S || z || e === _ || e === m || e === v || C || e === j || P || Z || Q || typeof e == "object" && e !== null && (e.$$typeof === T || e.$$typeof === y || e.$$typeof === N || e.$$typeof === A || e.$$typeof === h || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === $ || e.getModuleId !== void 0));
    }
    function be(e, r, t) {
      var i = e.displayName;
      if (i)
        return i;
      var u = r.displayName || r.name || "";
      return u !== "" ? t + "(" + u + ")" : t;
    }
    function b(e) {
      return e.displayName || "Context";
    }
    function x(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && n("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case R:
          return "Fragment";
        case O:
          return "Portal";
        case S:
          return "Profiler";
        case _:
          return "StrictMode";
        case m:
          return "Suspense";
        case v:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case A:
            var r = e;
            return b(r) + ".Consumer";
          case N:
            var t = e;
            return b(t._context) + ".Provider";
          case h:
            return be(e, e.render, "ForwardRef");
          case y:
            var i = e.displayName || null;
            return i !== null ? i : x(e.type) || "Memo";
          case T: {
            var u = e, l = u._payload, f = u._init;
            try {
              return x(f(l));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var L = Object.assign, U = 0, k, te, ce, oe, D, ve, de;
    function c() {
    }
    c.__reactDisabledLog = !0;
    function o() {
      {
        if (U === 0) {
          k = console.log, te = console.info, ce = console.warn, oe = console.error, D = console.group, ve = console.groupCollapsed, de = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: c,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        U++;
      }
    }
    function Y() {
      {
        if (U--, U === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: L({}, e, {
              value: k
            }),
            info: L({}, e, {
              value: te
            }),
            warn: L({}, e, {
              value: ce
            }),
            error: L({}, e, {
              value: oe
            }),
            group: L({}, e, {
              value: D
            }),
            groupCollapsed: L({}, e, {
              value: ve
            }),
            groupEnd: L({}, e, {
              value: de
            })
          });
        }
        U < 0 && n("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var V = W.ReactCurrentDispatcher, q;
    function K(e, r, t) {
      {
        if (q === void 0)
          try {
            throw Error();
          } catch (u) {
            var i = u.stack.trim().match(/\n( *(at )?)/);
            q = i && i[1] || "";
          }
        return `
` + q + e;
      }
    }
    var ie = !1, M;
    {
      var ae = typeof WeakMap == "function" ? WeakMap : Map;
      M = new ae();
    }
    function se(e, r) {
      if (!e || ie)
        return "";
      {
        var t = M.get(e);
        if (t !== void 0)
          return t;
      }
      var i;
      ie = !0;
      var u = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var l;
      l = V.current, V.current = null, o();
      try {
        if (r) {
          var f = function() {
            throw Error();
          };
          if (Object.defineProperty(f.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(f, []);
            } catch (B) {
              i = B;
            }
            Reflect.construct(e, [], f);
          } else {
            try {
              f.call();
            } catch (B) {
              i = B;
            }
            e.call(f.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (B) {
            i = B;
          }
          e();
        }
      } catch (B) {
        if (B && i && typeof B.stack == "string") {
          for (var a = B.stack.split(`
`), w = i.stack.split(`
`), p = a.length - 1, g = w.length - 1; p >= 1 && g >= 0 && a[p] !== w[g]; )
            g--;
          for (; p >= 1 && g >= 0; p--, g--)
            if (a[p] !== w[g]) {
              if (p !== 1 || g !== 1)
                do
                  if (p--, g--, g < 0 || a[p] !== w[g]) {
                    var I = `
` + a[p].replace(" at new ", " at ");
                    return e.displayName && I.includes("<anonymous>") && (I = I.replace("<anonymous>", e.displayName)), typeof e == "function" && M.set(e, I), I;
                  }
                while (p >= 1 && g >= 0);
              break;
            }
        }
      } finally {
        ie = !1, V.current = l, Y(), Error.prepareStackTrace = u;
      }
      var re = e ? e.displayName || e.name : "", Le = re ? K(re) : "";
      return typeof e == "function" && M.set(e, Le), Le;
    }
    function pe(e, r, t) {
      return se(e, !1);
    }
    function Je(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function he(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return se(e, Je(e));
      if (typeof e == "string")
        return K(e);
      switch (e) {
        case m:
          return K("Suspense");
        case v:
          return K("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case h:
            return pe(e.render);
          case y:
            return he(e.type, r, t);
          case T: {
            var i = e, u = i._payload, l = i._init;
            try {
              return he(l(u), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var ge = Object.prototype.hasOwnProperty, Se = {}, Pe = W.ReactDebugCurrentFrame;
    function ye(e) {
      if (e) {
        var r = e._owner, t = he(e.type, e._source, r ? r.type : null);
        Pe.setExtraStackFrame(t);
      } else
        Pe.setExtraStackFrame(null);
    }
    function Ge(e, r, t, i, u) {
      {
        var l = Function.call.bind(ge);
        for (var f in e)
          if (l(e, f)) {
            var a = void 0;
            try {
              if (typeof e[f] != "function") {
                var w = Error((i || "React class") + ": " + t + " type `" + f + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[f] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw w.name = "Invariant Violation", w;
              }
              a = e[f](r, f, i, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (p) {
              a = p;
            }
            a && !(a instanceof Error) && (ye(u), n("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", i || "React class", t, f, typeof a), ye(null)), a instanceof Error && !(a.message in Se) && (Se[a.message] = !0, ye(u), n("Failed %s type: %s", t, a.message), ye(null));
          }
      }
    }
    var He = Array.isArray;
    function Ee(e) {
      return He(e);
    }
    function Ke(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Xe(e) {
      try {
        return Fe(e), !1;
      } catch {
        return !0;
      }
    }
    function Fe(e) {
      return "" + e;
    }
    function Ne(e) {
      if (Xe(e))
        return n("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ke(e)), Fe(e);
    }
    var fe = W.ReactCurrentOwner, Ze = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, me, $e, Re;
    Re = {};
    function Qe(e) {
      if (ge.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Ce(e) {
      if (ge.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function ke(e, r) {
      if (typeof e.ref == "string" && fe.current && r && fe.current.stateNode !== r) {
        var t = x(fe.current.type);
        Re[t] || (n('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', x(fe.current.type), e.ref), Re[t] = !0);
      }
    }
    function De(e, r) {
      {
        var t = function() {
          me || (me = !0, n("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function Me(e, r) {
      {
        var t = function() {
          $e || ($e = !0, n("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var er = function(e, r, t, i, u, l, f) {
      var a = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: E,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: f,
        // Record the component responsible for creating this element.
        _owner: l
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: i
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: u
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function rr(e, r, t, i, u) {
      {
        var l, f = {}, a = null, w = null;
        t !== void 0 && (Ne(t), a = "" + t), Ce(r) && (Ne(r.key), a = "" + r.key), Qe(r) && (w = r.ref, ke(r, u));
        for (l in r)
          ge.call(r, l) && !Ze.hasOwnProperty(l) && (f[l] = r[l]);
        if (e && e.defaultProps) {
          var p = e.defaultProps;
          for (l in p)
            f[l] === void 0 && (f[l] = p[l]);
        }
        if (a || w) {
          var g = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && De(f, g), w && Me(f, g);
        }
        return er(e, a, w, u, i, fe.current, f);
      }
    }
    var _e = W.ReactCurrentOwner, Ie = W.ReactDebugCurrentFrame;
    function ee(e) {
      if (e) {
        var r = e._owner, t = he(e.type, e._source, r ? r.type : null);
        Ie.setExtraStackFrame(t);
      } else
        Ie.setExtraStackFrame(null);
    }
    var xe;
    xe = !1;
    function Te(e) {
      return typeof e == "object" && e !== null && e.$$typeof === E;
    }
    function Ae() {
      {
        if (_e.current) {
          var e = x(_e.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function nr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var We = {};
    function tr(e) {
      {
        var r = Ae();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function ze(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = tr(r);
        if (We[t])
          return;
        We[t] = !0;
        var i = "";
        e && e._owner && e._owner !== _e.current && (i = " It was passed a child from " + x(e._owner.type) + "."), ee(e), n('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, i), ee(null);
      }
    }
    function Ye(e, r) {
      {
        if (typeof e != "object")
          return;
        if (Ee(e))
          for (var t = 0; t < e.length; t++) {
            var i = e[t];
            Te(i) && ze(i, r);
          }
        else if (Te(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var u = X(e);
          if (typeof u == "function" && u !== e.entries)
            for (var l = u.call(e), f; !(f = l.next()).done; )
              Te(f.value) && ze(f.value, r);
        }
      }
    }
    function ir(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === h || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === y))
          t = r.propTypes;
        else
          return;
        if (t) {
          var i = x(r);
          Ge(t, e.props, "prop", i, e);
        } else if (r.PropTypes !== void 0 && !xe) {
          xe = !0;
          var u = x(r);
          n("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", u || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && n("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ar(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var i = r[t];
          if (i !== "children" && i !== "key") {
            ee(e), n("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", i), ee(null);
            break;
          }
        }
        e.ref !== null && (ee(e), n("Invalid attribute `ref` supplied to `React.Fragment`."), ee(null));
      }
    }
    function Ve(e, r, t, i, u, l) {
      {
        var f = d(e);
        if (!f) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var w = nr(u);
          w ? a += w : a += Ae();
          var p;
          e === null ? p = "null" : Ee(e) ? p = "array" : e !== void 0 && e.$$typeof === E ? (p = "<" + (x(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : p = typeof e, n("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", p, a);
        }
        var g = rr(e, r, t, u, l);
        if (g == null)
          return g;
        if (f) {
          var I = r.children;
          if (I !== void 0)
            if (i)
              if (Ee(I)) {
                for (var re = 0; re < I.length; re++)
                  Ye(I[re], e);
                Object.freeze && Object.freeze(I);
              } else
                n("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Ye(I, e);
        }
        return e === R ? ar(g) : ir(g), g;
      }
    }
    function sr(e, r, t) {
      return Ve(e, r, t, !0);
    }
    function fr(e, r, t) {
      return Ve(e, r, t, !1);
    }
    var ur = fr, lr = sr;
    le.Fragment = R, le.jsx = ur, le.jsxs = lr;
  }()), le;
}
process.env.NODE_ENV === "production" ? Oe.exports = or() : Oe.exports = vr();
var s = Oe.exports;
const dr = J.forwardRef(
  ({
    title: F,
    icon: E,
    shape: O,
    color: R = "#2196f3",
    isActive: _,
    index: S,
    currentStep: N,
    showProggressBar: A,
    layout: h,
    inlineStep: m = !1,
    darkColor: v,
    darkIconColor: y,
    removeBackgroundTab: T,
    removeBackgroundTabTransparentColor: j,
    showErrorOnTab: G,
    showErrorOnTabColor: H = "red",
    onClick: X
  }, W) => {
    const n = _ ? "active" : "", ne = O === "square" ? "default" : "", [P, Z] = J.useState(!1);
    J.useEffect(() => {
      _ && Z(!0);
    }, [_]);
    const Q = () => {
      const d = {
        border: "2px solid " + R
      };
      return v && (d.border = "2px solid " + v), G && (d.border = "2px solid " + H), h === "vertical" ? {
        ...d,
        rotate: "90deg",
        animation: "slideInVertical 0.3s forwards"
      } : d;
    }, C = () => {
      if (_ && y)
        return { color: y || R };
      if (_ && P)
        return { color: "white" };
      if (P && y)
        return { color: y || R };
      if (P)
        return { color: "white" };
    }, z = () => G && P && S <= N ? H : P && !T ? v || R : "";
    J.useImperativeHandle(W, () => ({
      setChecked: (d) => {
        Z(d);
      }
    }));
    const $ = () => E ? typeof E == "string" ? /* @__PURE__ */ s.jsx("i", { className: E, style: C() }) : E : /* @__PURE__ */ s.jsx("span", { style: C(), children: S + 1 });
    return /* @__PURE__ */ s.jsxs(
      "li",
      {
        className: `${n}`,
        style: {
          position: "relative"
        },
        children: [
          A && P && S <= N && /* @__PURE__ */ s.jsx(
            "div",
            {
              className: "smooth-border-left-to-right",
              style: Q()
            }
          ),
          /* @__PURE__ */ s.jsxs(
            "a",
            {
              className: `${_ ? "active" : ""} ${m ? "inline-step" : ""}`,
              style: { cursor: ne },
              onClick: X,
              children: [
                /* @__PURE__ */ s.jsx(
                  "div",
                  {
                    className: `wizard-icon-circle md ${P ? "checked" : ""} ${O === "square" ? "square_shape" : ""}`,
                    role: "tab",
                    tabIndex: _ ? 0 : void 0,
                    id: `step-${S}`,
                    "aria-controls": `step-${S}`,
                    "aria-disabled": _,
                    "aria-selected": _,
                    style: {
                      backgroundColor: T ? "transparent" : P ? v || R : "",
                      border: T ? "unset" : ""
                    },
                    children: /* @__PURE__ */ s.jsx(
                      "div",
                      {
                        className: `wizard-icon-container ${O === "square" ? "square_shape" : ""}`,
                        style: {
                          backgroundColor: z()
                        },
                        children: /* @__PURE__ */ s.jsx(
                          "span",
                          {
                            className: "wizard-icon",
                            style: T ? {
                              backgroundColor: j || "white",
                              padding: "10px"
                            } : {},
                            children: $()
                          }
                        )
                      }
                    )
                  }
                ),
                /* @__PURE__ */ s.jsx(
                  "span",
                  {
                    className: `stepTitle ${_ ? "active" : ""}`,
                    style: {
                      color: G && P && S <= N ? H : P ? v || R : "",
                      marginTop: m ? "" : "8px",
                      padding: m ? "0 10px" : "0"
                    },
                    children: F
                  }
                )
              ]
            }
          )
        ]
      },
      S
    );
  }
), we = ({
  darkTextColor: F,
  darkButtonColor: E,
  onClick: O,
  children: R
}) => /* @__PURE__ */ s.jsx(
  "button",
  {
    className: "wizard-btn",
    type: "button",
    style: {
      color: F,
      backgroundColor: E
    },
    onClick: O,
    children: R
  }
);
const pr = J.forwardRef(
  ({
    title: F,
    shape: E = "",
    color: O = "#2196f3",
    children: R,
    subtitle: _ = "",
    nextButtonText: S = "Next",
    nextButtonTemplate: N,
    backButtonText: A = "Back",
    backButtonTemplate: h,
    finishButtonText: m = "Finish",
    finishButtonTemplate: v,
    stepSize: y = "md",
    layout: T = "horizontal",
    startIndex: j = 0,
    disableBackOnClickStep: G = !1,
    showProggressBar: H = !0,
    inlineStep: X = !1,
    darkMode: W = !1,
    customDarkModeColor: n = {},
    //disable titles and subtitle color , background color and border color,buttons
    removeBackgroundTab: ne = !1,
    removeBackgroundTabTransparentColor: P = "",
    onComplete: Z,
    onTabChange: Q
  }, C) => {
    const z = J.Children.toArray(
      R
    ), $ = z.map(() => J.useRef(null));
    cr(C, () => ({
      nextTab: () => {
        U();
      },
      prevTab: () => {
        k();
      },
      reset: () => {
        x(j), $.forEach((c, o) => {
          var Y, V;
          j >= o ? (Y = c == null ? void 0 : c.current) == null || Y.setChecked(!0) : (V = c == null ? void 0 : c.current) == null || V.setChecked(!1);
        });
      },
      activeAll: () => {
        $.forEach((c) => {
          var o;
          (o = c == null ? void 0 : c.current) == null || o.setChecked(!0);
        });
      },
      goToTab: (c) => {
        L(c, !0), $.forEach((o, Y) => {
          var V, q;
          c >= Y ? (V = o == null ? void 0 : o.current) == null || V.setChecked(!0) : (q = o == null ? void 0 : o.current) == null || q.setChecked(!1);
        });
      }
    }));
    const [d, be] = Ue(!1);
    je(() => {
      W && be(!0);
    }, [W]), (j < 0 || j > z.length) && (j = 0, console.error(
      "startIndex should be greater than or equal to 0 or less than steps.length"
    ));
    const [b, x] = Ue(j);
    je(() => {
      b > 0 && $.forEach((c, o) => {
        var Y;
        j >= o && ((Y = c == null ? void 0 : c.current) == null || Y.setChecked(!0));
      });
    }, [b, j, $]), X && (H = !1), typeof Q == "function" && Q({
      prevIndex: b,
      nextIndex: b + 1
    });
    const L = (c, o = !1) => {
      if (o) {
        x(c);
        return;
      }
      c <= b && x(c);
    }, U = () => {
      b !== z.length - 1 && x(b + 1);
    }, k = () => {
      b !== 0 && x(b - 1);
    }, te = () => {
      typeof Z == "function" && Z();
    }, ce = () => z.map((c, o) => {
      const {
        title: Y,
        icon: V,
        isValid: q = !0,
        validationError: K,
        showErrorOnTab: ie,
        showErrorOnTabColor: M = "red"
      } = c.props, ae = o === b;
      return je(() => {
        var se, pe;
        ae && !q && (x(o - 1), (pe = (se = $[o]) == null ? void 0 : se.current) == null || pe.setChecked(!1), typeof K == "function" && K());
      }, [ae, q, o, K]), /* @__PURE__ */ s.jsx(
        dr,
        {
          ref: $[o],
          title: Y,
          icon: V,
          shape: E,
          color: O,
          isActive: ae,
          index: o,
          currentStep: b,
          layout: T,
          showProggressBar: H,
          inlineStep: X,
          darkColor: d && n.tab ? n.tab : "",
          darkIconColor: d && n.tabIconColor ? n.tabIconColor : "",
          removeBackgroundTab: ne,
          removeBackgroundTabTransparentColor: P,
          showErrorOnTab: ie,
          showErrorOnTabColor: M,
          onClick: () => G ? null : L(o)
        },
        o
      );
    }), oe = () => z[b], D = {
      backgroundColor: d && (n != null && n.buttons) ? n == null ? void 0 : n.buttons : O,
      borderColor: d && (n != null && n.buttons) ? n == null ? void 0 : n.buttons : O,
      color: d && (n != null && n.buttonsText) ? (n == null ? void 0 : n.buttonsText) + " !important" : "unset",
      borderRadius: "4px"
    }, ve = T === "vertical" ? "vertical" : "horizontal", de = X ? "inline" : "";
    return /* @__PURE__ */ s.jsxs(
      "div",
      {
        className: `react-form-wizard ${y} ${ve} ${de} `,
        children: [
          /* @__PURE__ */ s.jsx("div", { className: "wizard-header", children: typeof F == "string" ? /* @__PURE__ */ s.jsxs(s.Fragment, { children: [
            /* @__PURE__ */ s.jsx(
              "h4",
              {
                style: d && n.title ? { color: n.title } : {},
                className: "wizard-title",
                children: F
              }
            ),
            /* @__PURE__ */ s.jsx(
              "p",
              {
                style: d && n.subtitle ? { color: n.subtitle } : {},
                className: "category",
                children: _
              }
            )
          ] }) : F }),
          /* @__PURE__ */ s.jsxs("div", { className: "wizard-navigation", children: [
            /* @__PURE__ */ s.jsx(
              "ul",
              {
                className: `form-wizard-steps  wizard-nav wizard-nav-pills ${E} ${y}`,
                style: { borderColor: O },
                children: ce()
              }
            ),
            /* @__PURE__ */ s.jsx("div", { className: "wizard-tab-content", children: oe() })
          ] }),
          /* @__PURE__ */ s.jsxs("div", { className: "wizard-card-footer clearfix", children: [
            b > 0 && /* @__PURE__ */ s.jsx(s.Fragment, { children: h ? h(k) : /* @__PURE__ */ s.jsx("div", { className: "wizard-footer-left", style: D, children: /* @__PURE__ */ s.jsx(
              we,
              {
                darkTextColor: d && (n != null && n.buttonsText) ? n == null ? void 0 : n.buttonsText : "",
                darkButtonColor: d && (n != null && n.buttons) ? n == null ? void 0 : n.buttons : "",
                onClick: k,
                children: A
              }
            ) }) }),
            b < z.length - 1 && /* @__PURE__ */ s.jsx(s.Fragment, { children: N ? N(U) : /* @__PURE__ */ s.jsx("div", { className: "wizard-footer-right", style: D, children: /* @__PURE__ */ s.jsx(
              we,
              {
                darkTextColor: d && (n != null && n.buttonsText) ? n == null ? void 0 : n.buttonsText : "",
                darkButtonColor: d && (n != null && n.buttons) ? n == null ? void 0 : n.buttons : "",
                onClick: U,
                children: S
              }
            ) }) }),
            b === z.length - 1 && /* @__PURE__ */ s.jsx(s.Fragment, { children: v ? v(te) : /* @__PURE__ */ s.jsx("div", { className: "wizard-footer-right", style: D, children: /* @__PURE__ */ s.jsx(
              we,
              {
                darkTextColor: d && (n != null && n.finishButtonText) ? n == null ? void 0 : n.finishButtonText : "",
                darkButtonColor: d && (n != null && n.finishButton) ? n == null ? void 0 : n.finishButton : "",
                onClick: te,
                children: m
              }
            ) }) })
          ] })
        ]
      }
    );
  }
);
pr.TabContent = ({ children: F, isValid: E = !0 }) => /* @__PURE__ */ s.jsx(s.Fragment, { children: E && F });
export {
  pr as default
};
